### ISSUE TITLE
Compiling

### ISSUE THREAD

**Issue by demipixel** [PrismarineJS/flying-squid#50]
Hopefully many people will be using this, which means lots of people unfamiliar with node.js will be using it. [Something to do this](http://enclosejs.com/) exists however there are a few things to note:
- [Look at the bottom](https://github.com/igorklopov/enclose/wiki/02.Packaging). We might need to do special stuff with FS and account for that!
- Cross compilation is not supported

So ideally, something like auto-squid is running but instead of pulling and then putting a server up, it pulls, runs encloseJS, puts the compiled version in a different folder, and then pushes it to ANOTHER repo (so that github is hosting it, not us).

So yeah, if we wanted to do this, the biggest issues are the file system and, more importantly, cross compilation (unless mac servers exist or people are willing to host on a mac).

The main reason I'm talking about fs is because
- It gets used in serverPlugins.js
- The compiled version would need to access a "plugins" folder in the same folder.

So there might be a plugin that auto-compile-squid needs to add that would run external plugins from OUTSIDE the compiled version.

**TL;DR It's not an emergency or anything but it would be cool to have and useful for 50% of our users**

**Comment by rom1504** [comment]
So the idea is to distribute binaries including everything for each operating system ? do people actually want to run mc server on anything than linux ?
Also see  #15

**Comment by demipixel** [comment]
Doesn't hurt. Otherwise we can just make it a repo of people with contributors who are willing to just compile once a week or something...

**Comment by rom1504** [comment]
On that theme : https://medium.com/@iefserge/runtime-js-javascript-library-os-823ada1cc3c

**Comment by demipixel** [comment]
To use encloseJS (doesn't require the installation of anything unlike runtimejs), it doesn't want to use variables in require()s (i.e. literal strings only). MMMmmmmm... I'm getting errors with QEMU too (the runtimejs thing)

**Comment by rom1504** [comment]
let's not do that. we can go docker if needed
